dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -288,24 +306,12 @@ void Microsoft::Windows::AppNotifications::Helpers::RegisterAssets(std::wstring
         &hKey,
         nullptr /* lpdwDisposition */));
 
-    // Try the following techniques to retrieve display name and icon:
-    // 1. Based on the best app shortcut, using the FrameworkUdk.
-    // 2. From the current process.
-    // 3. Set a default DisplayName, but leave empty the icon file path so Shell can set a default icon.
-    Microsoft::Windows::AppNotifications::ShellLocalization::AppNotificationAssets assets{};
-
-    if (FAILED(Microsoft::Windows::AppNotifications::ShellLocalization::RetrieveAssetsFromShortcut(assets)) &&
-        FAILED(Microsoft::Windows::AppNotifications::ShellLocalization::RetrieveAssetsFromProcess(assets)))
-    {
-        assets.displayName = GetDisplayNameBasedOnProcessName();
-    }
-
-    RegisterValue(hKey, L"DisplayName", reinterpret_cast<const BYTE*>(assets.displayName.c_str()), REG_EXPAND_SZ, assets.displayName.size() * sizeof(wchar_t));
+    RegisterValue(hKey, L"DisplayName", reinterpret_cast<const BYTE*>(displayName.c_str()), REG_EXPAND_SZ, displayName.size() * sizeof(wchar_t));
 
     // If no icon is specified in the Registry, the OS will render a default icon for App Notifications.
-    if (!assets.iconFilePath.empty())
+    if (!iconFilePath.empty())
     {
-        RegisterValue(hKey, L"IconUri", reinterpret_cast<const BYTE*>(assets.iconFilePath.c_str()), REG_EXPAND_SZ, assets.iconFilePath.size() * sizeof(wchar_t));
+        RegisterValue(hKey, L"IconFilePath", reinterpret_cast<const BYTE*>(iconFilePath.c_str()), REG_EXPAND_SZ, iconFilePath.size() * sizeof(wchar_t));
```

Comment:
@danielayala94 - does the shell expect the parameter to iconUri or iconFilePath? While testing it did not make a difference. Let me know if you are aware?

Reply:
It is IconUri, that's how Shell knows it.